engine/core/src/main/java/org/datacleaner/job/runner/AbstractRowProcessingConsumer.java

```diff
@@ -121,7 +119,7 @@ private boolean isAlwaysSatisfiedRequirement() {
         final Set<Object> sourceJobsOfInputColumns = sourceColumnFinder.findAllSourceJobs(inputColumnSinkJob);
         for (Iterator<Object> it = sourceJobsOfInputColumns.iterator(); it.hasNext();) {
             final Object sourceJob = it.next();
-            if (sourceJob instanceof HasComponentRequirement) {
+            if (sourceJob instanceof HasComponentRequirement && sourceJob instanceof InputColumnSourceJob) {
```

Comment:
For reference - this change was what made the integration test finally pass ... It seems that `buildSourceJobsOfInputColumns` was creating a list with any kind of source component. The naming indicates however that it's only sources of columns (so transformers).